tools/autograd/derivatives.yaml

```diff
@@ -1261,53 +1291,66 @@
 
 - name: t(Tensor(a) self) -> Tensor(a)
   self: grad.t()
+  result: auto_linear
 
 - name: one_hot(Tensor self, int num_classes=-1) -> Tensor
   self: non_differentiable
 
 - name: flip(Tensor self, int[] dims) -> Tensor
   self: grad.flip(dims)
+  result: auto_linear
 
 - name: roll(Tensor self, int[1] shifts, int[1] dims=[]) -> Tensor
   self: grad.roll(fmap(reverse_list(shifts), [](int64_t i){return -i;}), reverse_list(dims))
+  result: auto_linear
 
 - name: rot90(Tensor self, int k=1, int[] dims=[0,1]) -> Tensor
   self: grad.rot90(-k, dims)
+  result: auto_linear
 
 - name: take(Tensor self, Tensor index) -> Tensor
   self: zeros_like(self).put_(index, grad, true)
   index: non_differentiable
+  result: auto_linear
 
 - name: tan(Tensor self) -> Tensor
   self: grad * (1 + result.pow(2)).conj()
+  result: auto_element_wise
 
 - name: tanh(Tensor self) -> Tensor
   self: tanh_backward(grad, result)
+  result: auto_element_wise
 
 - name: topk(Tensor self, int k, int dim=-1, bool largest=True, bool sorted=True) -> (Tensor values, Tensor indices)
   self: value_selecting_reduction_backward(grad, dim, indices, self.sizes(), true)
   output_differentiability: [True, False]
 
 - name: trace(Tensor self) -> Tensor
   self: trace_backward(grad, self.sizes())
+  result: auto_linear
 
 - name: transpose.int(Tensor(a) self, int dim0, int dim1) -> Tensor(a)
   self: grad.transpose(dim0, dim1)
+  result: auto_linear
 
 - name: transpose_(Tensor(a!) self, int dim0, int dim1) -> Tensor(a!)
   self: grad.transpose(dim0, dim1)
+  result: self_t.transpose(dim0, dim1)
 
 - name: triangular_solve(Tensor self, Tensor A, bool upper=True, bool transpose=False, bool unitriangular=False) -> (Tensor solution, Tensor cloned_coefficient)
   self, A: triangular_solve_backward(grads[0], grads[1], self, A, solution, upper, transpose, unitriangular, grad_input_mask)
 
 - name: tril(Tensor self, int diagonal=0) -> Tensor
   self: grad.tril(diagonal)
+  result: auto_linear
 
 - name: triu(Tensor self, int diagonal=0) -> Tensor
   self: grad.triu(diagonal)
+  result: auto_linear
```

Comment:
Can these be auto_element_wise?

Reply:
They can, but the linear version is more efficient, so I prefer it when possible.